### PR TITLE
Con 4187 save cohort send to employer

### PR DESCRIPTION
**Pull request by VasanthaKasirajan3008**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddAndApproveDraftApprenticeshipsViewModelMapper.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            var viewModel = new BulkUploadAddAndApproveDraftApprenticeshipsViewModel
+            {
+                BulkUploadDraftApprenticeshipsViewModel = (from result in source.BulkUploadAddAndApproveDraftApprenticeshipResponse
+                                                           let bulkUploadDraftApprenticeshipsViewModel = new BulkUploadDraftApprenticeshipViewModel
+                                                           {
+                                                               CohortReference = result.CohortReference,
+                                                               NumberOfApprenticeships = result.NumberOfApprenticeships,
+                                                               EmployerName = result.EmployerName
+                                                           }
+                                                           select bulkUploadDraftApprenticeshipsViewModel).ToList()
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddAndApproveDraftApprenticeshipsViewModelMapper.cs:22]:
```suggestion
                BulkUploadDraftApprenticeshipsViewModel = source.BulkUploadAddAndApproveDraftApprenticeshipResponse.Select(r => new BulkUploadDraftApprenticeshipViewModel 
{
                CohortReference = r.CohortReference,
                NumberOfApprenticeships = r.NumberOfApprenticeships,
                EmployerName = r.EmployerName
 }).ToList()
```

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly IEncodingService _encodingService;
+        public FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper(ICacheService cacheService, IEncodingService encodingService)
+        {
+            _cacheService = cacheService;
+            _encodingService = encodingService;
+        }
+
+        public async Task<BulkUploadAddAndApproveDraftApprenticeshipsRequest> Map(FileUploadReviewViewModel source)
+        {
+            var csVRecrods = await _cacheService.GetFromCache<List<CsvRecord>>(source.CacheRequestId.ToString());
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs:24]: typo `csvRecords`

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            await _cacheService.ClearCache(source.CacheRequestId.ToString());
+            return new BulkUploadAddAndApproveDraftApprenticeshipsRequest
+            {
+                ProviderId = source.ProviderId,
+                BulkUploadAddAndApproveDraftApprenticeships = csVRecrods.Select(x => MapTo(x))
```
Comment (cofaulco) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/FileUploadReviewViewModelToBulkUploadAddAndApproveDraftApprenticeshipsRequestMapper.cs:29]: can be simplified to 
```suggestion
                BulkUploadAddAndApproveDraftApprenticeships = csVRecrods.Select(MapTo)
```

**Review by cofaulco — CHANGES_REQUESTED** [review]
couple of suggestions

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddAndApproveDraftApprenticeshipsViewModelMapper.cs (hunk 1 above)
Comment (VasanthaKasirajan3008) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Cohort/BulkUploadAddAndApproveDraftApprenticeshipsViewModelMapper.cs]: done

**Review by cofaulco — APPROVED** [review]

**Pull request merged**